web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx

```diff
@@ -0,0 +1,168 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Danger } from 'design/Alert';
+import Flex from 'design/Flex';
+import Indicator from 'design/Indicator';
+import { useEffect } from 'react';
+import { useAsync } from 'shared/hooks/useAsync';
+import { useTheme } from 'styled-components';
+import { H1 } from 'design/Text';
+import Box from 'design/Box';
+import { H3, P, P3 } from 'design/Text/Text';
+import { ButtonSecondary } from 'design/Button';
+import Image from 'design/Image';
+
+import { State as ResourcesState } from 'teleport/components/useResources';
+import { Role, RoleWithYaml } from 'teleport/services/resources';
+import { yamlService } from 'teleport/services/yaml';
+import { YamlSupportedResourceKind } from 'teleport/services/yaml/types';
+import { ButtonLockedFeature } from 'teleport/components/ButtonLockedFeature';
+
+import { RoleEditor } from './RoleEditor';
+import tagpromo from './tagpromo.png';
+
+/**
+ * This component is responsible for converting from the `Resource`
+ * representation of a role to a more accurate `RoleWithYaml` structure. The
+ * conversion is asynchronous and it's performed on the server side.
+ */
+export function RoleEditorAdapter({
+  resources,
+  onSave,
+  onDelete,
+  onCancel,
+}: {
+  resources: ResourcesState;
+  onSave: (role: Partial<RoleWithYaml>) => Promise<void>;
+  onDelete: () => Promise<void>;
+  onCancel: () => void;
+}) {
+  const theme = useTheme();
+  const [convertAttempt, convertToRole] = useAsync(
+    async (yaml: string): Promise<RoleWithYaml | null> => {
+      if (resources.status === 'creating' || !resources.item) {
+        return null;
+      }
+      return {
+        yaml,
+        object: await yamlService.parse<Role>(YamlSupportedResourceKind.Role, {
+          yaml,
+        }),
+      };
+    }
+  );
+
+  const originalContent = resources.item?.content ?? '';
+  useEffect(() => {
+    convertToRole(originalContent);
+  }, [originalContent]);
+
+  return (
+    <Flex flex="1">
+      <Flex
+        flexDirection="column"
+        borderLeft={1}
+        borderColor={theme.colors.interactive.tonal.neutral[0]}
+        backgroundColor={theme.colors.levels.surface}
+        width="700px"
+      >
+        {convertAttempt.status === 'processing' && (
+          <Flex
+            flexDirection="column"
+            alignItems="center"
+            justifyContent="center"
+            flex="1"
+          >
+            <Indicator />
```

Comment:
My linter is throwing an error saying `default is deprecated`. 
Looks like you can import the export directly instead of importing as default
```
import { Indicator } from 'design/Indicator';
```